### ISSUE TITLE
A11y addon freezes the UI

### ISSUE THREAD

**Issue by jsomsanith** [storybooks/storybook#5973]
**Describe the bug**
With the a11y addon, with complex stories, the UI blocks, making the user experience awful.
For some cases, switching from story to story takes 3 to 5 seconds, freezing the UI.

**To Reproduce**
1. Create a storybook with complex markup
2. Add the a11y addon
3. Switch from stories to stories
4. The render takes a lot of time

An example is published [here](http://2087.talend.surge.sh/forms/?path=/story/core-concepts--core-conditional-render)

**Expected behavior**
Render of stories should be faster.

**Screenshots**
![Mar-08-2019 22-08-15](https://user-images.githubusercontent.com/10761073/54056018-182b7080-41ef-11e9-8278-636449153d9f.gif)

**System:**
 - OS: MacOS Mojave
 - Device: Macbook Pro 2018
 - Browser: chrome
 - Framework: react
 - Addons: a11y
 - Version: 5.0.0

**Additional context**
I run the chrome devtool flamechart, to discover that a11y run is done at each story render, blocking the main thread.
Even if the a11y addon tab is not open, it runs. It's a waist of time because I don't even look at the reports at this moment.

What I propose : the run should be done at render time (STORY_RENDERED event) if the tab is active. Otherwise, it should run only when tab becomes active (REQUEST event).

**Comment by jsomsanith** [comment]
(I can contribute on that, I just need the community and maintainers feedback)

**Comment by shilman** [comment]
@jsomsanith This sounds super reasonable. Need any help getting set up to develop storybook?

**Comment by jsomsanith** [comment]
@shilman Thank you for your feedback.
Don't know yet for the set up, I'll follow any guide I find on the repo, if I have any struggle, I'll ping you :)

**Comment by shilman** [comment]
Yippee!! I just released https://github.com/storybooks/storybook/releases/tag/v5.1.0-alpha.6 containing PR #6044 that references this issue. Upgrade today to try it out!

Because it's a pre-release you can find it on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.
